src/pygeometadata/mcf.py

```diff
@@ -0,0 +1,413 @@
+                f'received {type(keywords)} instead')
+
+        section_dict = {
+            'keywords': keywords,
+            'keywords_type': keywords_type
+        }
+
+        if vocabulary:
+            section_dict['vocabulary'] = vocabulary
+        self.mcf['identification']['keywords'][section] = section_dict
+
+    def describe_band(self, band_number, name=None, title=None, abstract=None,
+                      units=None):
+        """Define metadata for a raster band.
+
+        Args:
+            band_number (int): a raster band index, starting at 1
+            name (str): name for the raster band
+            title (str): title for the raster band
+            abstract (str): description of the raster band
+            units (str): unit of measurement for the band's pixel values
+        """
+        idx = band_number - 1
+        attribute = self.mcf['content_info']['attributes'][idx]
+        if name is not None:
+            attribute['name'] = name
+        if title is not None:
+            attribute['title'] = title
+        if abstract is not None:
+            attribute['abstract'] = abstract
+        if units is not None:
+            attribute['units'] = units
+
+        self.mcf['content_info']['attributes'].insert(idx, attribute)
+
+    def describe_field(self, name, title=None, abstract=None,
+                       units=None):
+        """Define metadata for a tabular field.
+        Args:
+            name (str): name and unique identifier of the field
+            title (str): title for the field
+            abstract (str): description of the field
+            units (str): unit of measurement for the field's values
+        """
+        def get_attr(attribute_list):
+            for idx, attr in enumerate(attribute_list):
+                if attr['name'] == name:
+                    return idx, attr
+            raise ValueError(
+                f'{self.datasource} has no attribute named {name}')
+
+        idx, attribute = get_attr(self.mcf['content_info']['attributes'])
+
+        if title is not None:
+            attribute['title'] = title
+        if abstract is not None:
+            attribute['abstract'] = abstract
+        if units is not None:
+            attribute['units'] = units
+
+        self.mcf['content_info']['attributes'].insert(idx, attribute)
+
+    def write(self):
+        """Write MCF to disk."""
+        with open(self.mcf_path, 'w') as file:
+            file.write(yaml.dump(self.mcf))
+        # TODO: always also write an ISO-191* XML doc?
+
+    def validate(self):
+        """Validate MCF against a jsonschema object."""
+        # validate against our own schema, which could
+        # be a superset of the core MCF schema.
+        # If we wanted to validate against core MCF,
+        # we could use pygeometa.core.validate_mcf
+        jsonschema.validate(self.mcf, MCF_SCHEMA)
+
+    def to_string(self):
+        pass
+
+    def get_spatial_info(self):
```

Comment:
Yes, thanks for this suggestion. I also renamed it to `_set_spatial_info` since that is much more accurate.

I also changed the `__init__` logic to always call this method, even when an MCF already exists on disk, just in case the dataset was modified since the MCF was created.